File: 01-Languages/PHP/05-Control-Flow/Solutions/solution.php
<?php
// Solution to the FizzBuzz exercise using match (true) { ... }.

function fizzbuzz(int $n): string {
    return match (true) {
        $n % 15 === 0 => "FizzBuzz",
        $n % 3 === 0 => "Fizz",
        $n % 5 === 0 => "Buzz",
        default => (string) $n,
    };
}

for ($i = 1; $i <= 15; $i++) {
    echo fizzbuzz($i) . "\n";
}

File: 01-Languages/Perl/05-Control-Flow/Solutions/solution.pl
#!/usr/bin/env perl
use strict;
use warnings;

# 1. FizzBuzz
for my $n (1..20) {
    if ($n % 15 == 0) { print "FizzBuzz\n"; }
    elsif ($n % 3 == 0) { print "Fizz\n"; }
    elsif ($n % 5 == 0) { print "Buzz\n"; }
    else { print "$n\n"; }
}

# 2. Grades
my @scores = (95, 82, 71, 40);
for my $score (@scores) {
    my $grade;
    if    ($score >= 90) { $grade = "A"; }
    elsif ($score >= 80) { $grade = "B"; }
    elsif ($score >= 70) { $grade = "C"; }
    else                 { $grade = "F"; }
    print "$score -> $grade\n";
}

# 3. Countdown with until
my $n = 5;
until ($n < 1) {
    print "countdown: $n\n";
    $n--;
}
